book/i18n-using-yandex-as-translation-source.md

```diff
@@ -0,0 +1,92 @@
+Using machine translation (such as google or yandex) as translation source.
+================================
+
+Sometimes we need to translate the site into other languages.
+I do not recommend to use machine translation as production ready solution, but it can help to prepare the site to the translation by professional translators.
+
+
+## How to do it
+
+First of all we need catch MissingTranslationEvent. It is more convenient to wrap catch method in component.
+We use DbMessageSource to store traslated messages (the appropriate database migration is needed).
+
+```php
+namespace common\components;
+
+use yii\helpers\ArrayHelper;
+use yii\i18n\MissingTranslationEvent;
+
+class MachineTranslation extends \yii\base\Component
+{
+        /**
+         * Provider api key.
+         */
+        public $apiKey;
+
+        /**
+         * @param MissingTranslationEvent $event
+         */
+        public function handleMissingTranslation(MissingTranslationEvent $event)
+        {
+                /* @var $messageSource \yii\i18n\DbMessageSource */
+                $messageSource = $event->sender;
+                $db = $messageSource->db;
+
+                $sourceLang = explode('-', $messageSource->sourceLanguage)[0];
+                $targetLang = explode('-', $event->language)[0];
+        
+                $content = file_get_contents('https://translate.yandex.net/api/v1.5/tr.json/translate?' . http_build_query([
```

Comment:
Since Yandex URL is hardcoded, it's better to call the class `YandexTranslation`.